tests/scenarios/v2-addon-dev-test.ts

```diff
@@ -0,0 +1,209 @@
+              ]),
+
+              addon.hbs(),
+              addon.dependencies(),
+
+              babel({ babelHelpers: 'bundled' }),
+
+              addon.clean(),
+            ],
+          };
+        `,
+        components: {
+          demo: {
+            'button.hbs': `
+              <button {{on 'click' @onClick}}>
+                flip
+              </button>
+            `,
+            'out.hbs': `
+              <out>{{yield}}</out>
+            `,
+            'index.js': `
+                import Component from '@glimmer/component';
+
+                import FlipButton from './button';
+                import BlahButton from './button.hbs';
+                import Out from './out';
+
+                export default class ExampleComponent extends Component {
+                  Button = FlipButton;
+                  Button2 = BlahButton;
+                  Out = Out;
+
+                  @tracked active = false;
+
+                  flip = () => (this.active = !this.active);
+                }
+              `,
+            'index.hbs': `
+              Hello there!
+
+              <this.Out>{{this.active}}</this.Out>
+
+              <this.Button @onClick={{this.flip}} />
+              <this.Button2 @onClick={{this.flip}} />
+            `,
+          },
+        },
+      },
+    });
+
+    addon.linkDependency('@embroider/addon-shim', { baseDir: __dirname });
+    addon.linkDependency('@embroider/addon-dev', { baseDir: __dirname });
+
+    await addon.write();
+    let addonDir = addon.baseDir;
+
+    await execa('yarn', ['install'], { cwd: addonDir });
+    await execa('yarn', ['rollup', '-c', './rollup.config.js'], { cwd: addonDir });
+
+    project.addDevDependency(addon);
+
+    merge(project.files, {
+      tests: {
+        'the-test.js': `
+          import { click, render } from '@ember/test-helpers';
+          import { hbs } from 'ember-cli-htmlbars';
+          import { module, test } from 'qunit';
+          import { setupRenderingTest } from 'ember-qunit';
+
+          module('v2 addon tests', function (hooks) {
+            setupRenderingTest(hooks);
+
+            test('<Demo />', async function (assert) {
+              await render(hbs\`<Demo />\`);
+
+              assert.dom('out').containsText('false');
+
+              await click('button');
+
+              assert.dom('out').containsText('true');
+            });
+
+            test('<Demo::Out />', async function (assert) {
+              await render(hbs\`<Demo::Out>hi</Demo::Out>\`);
+
+              assert.dom('out').containsText('hi');
+            });
+          });
+        `,
+      },
+    });
+  })
+  .forEachScenario(scenario => {
+    Qmodule(scenario.name, function (hooks) {
+      let app: PreparedApp;
+      hooks.before(async () => {
+        app = await scenario.prepare();
+      });
+
+      Qmodule('The addon', function (hooks) {
+        let addonDir = '';
+        hooks.before(async () => {
+          debugger;
```

Comment:
This is temporary, because I can't trust my work computer atm with the registry errors I'm seeing from the yarn install.

If there is a different way I should install deps with a fixturify project, lemme know

Reply:
To add a dependency

1. Add it to tests/scenarios/package.json so that it will be visible to tests/scenarios/v2-addon-dev-test.ts. You can use aliasing to make multiple versions available as necessary.
2. Add it to a specific `project` using `project.linkDevDependency` or `project.linkDependency`.

Reply:
it this part, I'm just wanting the addon's location on disk so I can assert that the package.json changed successfully.
I only have access to `scenario` and `app` at this location (of the relevant objects anyway), so once I get through your other suggestions, I'll see if there is a way to get that info

Reply:
resolved by storing `addon` in module-scope for later access in the tests